### PR TITLE
Add headless support on macOS

### PR DESCRIPTION
**Pull request by ligfx**

### CODE REVIEW

File: Source/Core/Common/GL/GLInterface/AGL.mm [hunk 1]
```diff
@@ -49,15 +33,34 @@
     return false;
   }
 
-  if (cocoaWin == nil)
+  if (window_handle)
```
Comment (stenzek) [inline Source/Core/Common/GL/GLInterface/AGL.mm]: Could do if (!window_handle) return true; here

Comment (ligfx) [inline Source/Core/Common/GL/GLInterface/AGL.mm]: I sort of like the visual distinction the indent provides, it's easy to see which part is only used with a window. What do you think?

Comment (stenzek) [inline Source/Core/Common/GL/GLInterface/AGL.mm]: I don't really mind either way, it's just the usual style followed here AFAIK.

**Comment by stenzek** [comment]
LGTM

File: Source/Core/Common/GL/GLInterface/AGL.mm (hunk 1 above)
Comment (lioncash) [inline Source/Core/Common/GL/GLInterface/AGL.mm]: Yeah, the general style is to situate early-return cases first. It makes it much nicer to read (conceptually) as it places invalid code and any necessary handling first before actual behavior. If an indentation makes code nicer to read, that's usually a signifier that the contained code should be moved to another function, or it's too intermixed with other stuff.

Comment (ligfx) [inline Source/Core/Common/GL/GLInterface/AGL.mm]: I refactored a bit to move code into other functions.

File: Source/Core/Common/GL/GLInterface/AGL.mm [hunk 2]
```diff
@@ -5,31 +5,54 @@
 #include "Common/GL/GLInterface/AGL.h"
 #include "Common/Logging/Log.h"
 
-void cInterfaceAGL::Swap()
+static bool DimensionsWereUpdated(NSView* view, u32* width, u32* height)
```
Comment (lioncash) [inline Source/Core/Common/GL/GLInterface/AGL.mm]: This should likely have a better name. This currently reads as something that solely checks values but doesn't modify them at call sites.

**Comment by Orphis** [comment]
Good, I'm going to continue thinking of features I want, not say a word and expect @ligfx to implement them for me. Thanks!

**Comment by ligfx** [comment]
@Orphis I got you fam 👍

**Comment by ligfx** [comment]
Ping!

Also @Sonicadvance1 I know I talked with you on IRC, but there is anything else about this that I need to change?

**Comment by shuffle2** [comment]
is there something pending here?

**Comment by ligfx** [comment]
@shuffle2 Not that I know of.

**Pull request merged**